src/app/modules/main/module.nim

```diff
@@ -494,9 +494,22 @@ method communityJoined*[T](
   self.communitySectionsModule[community.id].load(events, settingsService, contactsService, chatService, communityService, messageService)
 
   let communitySectionItem = self.createCommunityItem(community)
-  self.view.addItem(self.createCommunityItem(community))
+  self.view.addItem(communitySectionItem)
   self.setActiveSection(communitySectionItem)
 
+method communityLeft*[T](self: Module[T], communityId: string) =
+  if(not self.communitySectionsModule.contains(communityId)):
+    echo "main-module, unexisting community key to leave: ", communityId
+    return
+
+  self.communitySectionsModule.del(communityId)
+
+  self.view.removeItem(communityId)
```

Comment:
@jrainville you could access the model directly and do this:
`self.view.model().removeItem(communityId)`
that would save you adding procs to the view.